source/common/json/json_streamer.cc

```diff
@@ -1,199 +1,3 @@
-  ASSERT(levels_.top() == level);
-  levels_.pop();
-}
-
-void Streamer::push(Level* level) { levels_.push(level); }
-#endif
-
-void Streamer::Level::nextField() {
-  if (is_first_) {
-    is_first_ = false;
-  } else {
-    streamer_.addConstantString(",");
-  }
-}
-
-void Streamer::Map::nextField() {
-  if (expecting_value_) {
-    expecting_value_ = false;
-  } else {
-    Level::nextField();
-  }
-}
-
-void Streamer::Map::addKey(absl::string_view key) {
-  ASSERT_THIS_IS_TOP_LEVEL;
-  ASSERT(!expecting_value_);
-  nextField();
-  streamer_.addSanitized("\"", key, "\":");
-  expecting_value_ = true;
-}
-
-void Streamer::Map::addEntries(const Entries& entries) {
-  for (const NameValue& entry : entries) {
-    addKey(entry.first);
-    addValue(entry.second);
-  }
-}
-
-void Streamer::Level::addValue(const Value& value) {
-  switch (value.index()) {
-  case 0:
-    static_assert(std::is_same<decltype(absl::get<0>(value)), const absl::string_view&>::value,
-                  "value at index 0 must be an absl::string_vlew");
-    addString(absl::get<absl::string_view>(value));
-    break;
-  case 1:
-    static_assert(std::is_same<decltype(absl::get<1>(value)), const double&>::value,
-                  "value at index 1 must be a double");
-    addNumber(absl::get<double>(value));
-    break;
-  case 2:
-    static_assert(std::is_same<decltype(absl::get<2>(value)), const uint64_t&>::value,
-                  "value at index 2 must be a uint64_t");
-    addNumber(absl::get<uint64_t>(value));
-    break;
-  case 3:
-    static_assert(std::is_same<decltype(absl::get<3>(value)), const int64_t&>::value,
-                  "value at index 3 must be an int64_t");
-    addNumber(absl::get<int64_t>(value));
-    break;
-  case 4:
-    static_assert(std::is_same<decltype(absl::get<4>(value)), const bool&>::value,
-                  "value at index 4 must be a bool");
-    addBool(absl::get<bool>(value));
-    break;
-  default:
-    IS_ENVOY_BUG(absl::StrCat("addValue invalid index: ", value.index()));
-    break;
-  }
-}
-
-void Streamer::Array::addEntries(const Entries& values) {
-  for (const Value& value : values) {
-    addValue(value);
-  }
-}
-
-void Streamer::addNumber(double number) {
-  if (std::isnan(number)) {
-    response_.addFragments({"null"});
-  } else {
-    Buffer::Util::serializeDouble(number, response_);
-  }
-}
-
-void Streamer::addNumber(uint64_t number) { response_.addFragments({absl::StrCat(number)}); }
-
-void Streamer::addNumber(int64_t number) { response_.addFragments({absl::StrCat(number)}); }
-
-void Streamer::addBool(bool b) { response_.addFragments({b ? "true" : "false"}); }
-
-void Streamer::addSanitized(absl::string_view prefix, absl::string_view str,
-                            absl::string_view suffix) {
-  absl::string_view sanitized = Json::sanitize(sanitize_buffer_, str);
-  response_.addFragments({prefix, sanitized, suffix});
-}
-
-} // namespace Json
+namespace Json {} // namespace Json
```

Comment:
the way you structured this PR Is maybe a better end result but is hard to review.

Can you follow how I did this in #35647 -- that shows what functionality actually needed to change.

A follow-up could be done after that which is completely non-functional to move the contents of the cc file into the .h.

Reply:
Okay

Reply:
Okay